neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jResultSet.java

```diff
@@ -60,6 +63,11 @@
 
 	private static final List<String> ACCEPTED_TYPES_FOR_FLATTENING = Arrays.asList("NODE", "RELATIONSHIP");
 
+	private static final Pattern SPECIAL_CHARS = Pattern.compile("[^a-z0-9]", Pattern.CASE_INSENSITIVE);
```

Comment:
misses underscore

also perhaps a positive pattern would be better? i.e. `^[a-z_][a-z0-9_]+$` + case insentistive ?